ts/private/ts_project.bzl

```diff
@@ -316,9 +329,9 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
             },
         )
 
-    transitive_sources = js_lib_helpers.gather_transitive_sources(output_sources, srcs_tsconfig_deps)
```

Comment:
This is the main change for the fix, not gathering sources+types from the `tsconfig` target as outputs.